### DISCUSSION TITLE
Cannot use readStructArray for codecs not registered  in OpcUaDefaultBinaryEncoding.ENCODING_NAME namespace

### DISCUSSION THREAD

**Discussion by opaquot** [eclipse-milo/milo#1336] [General]
**Describe the bug**
While trying to read an array of structure whose codecs are registered in another namespace index than 0, I cannot use the readStructArray method because it is looking for codecs only in namespace 0 (see [here](https://github.com/eclipse/milo/blob/master/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/serialization/OpcUaBinaryStreamDecoder.java#L1194))

**Expected behavior**
I should be able to read an array of custom structure whose codecs has been registered.

**Additional context**
Version of Eclipse Milo you are using: 0.6.14

**Comment by kevinherron** [comment]
The code you referred to is about the DataType _encoding_, which is not the same thing as the codec. Arrays of custom structures from namespaces other than index 0 do work.

More likely there is a problem reading the DataType Dictionary that contains that structure.

If you can share more details and set the logging level for the logger in `DataTypeDictionaryReader` to TRACE we can try to troubleshoot your issue.
